linker/Tests/Mono.Linker.Tests.Cases/Reflection/ConstructorUsedViaReflection.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Reflection;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+
+namespace Mono.Linker.Tests.Cases.Reflection
+{
+	public class ConstructorUsedViaReflection {
+		public static void Main ()
+		{
+			var constructor = typeof (OnlyUsedViaReflection).GetConstructor (BindingFlags.Public, null, new Type[]{}, new ParameterModifier[]{});
+			constructor.Invoke (null, new object[] { });
+		}
+
+		[Kept]
+		private class OnlyUsedViaReflection {
+			[Kept]
+			public OnlyUsedViaReflection ()
+			{ }
+
+			[Kept]
+			public OnlyUsedViaReflection(string bar)
+			{ }
+
+			private OnlyUsedViaReflection (int foo)
```

Comment:
please also add protected and internal cases

Reply:
I've added these.